### PR TITLE
Remove Content

### LINKED ISSUE #1380: [2.0] Remove posts and comments

### PR DESCRIPTION
**Pull request by Sjmarf**
Allows moderators to remove posts and comments. Relies on [this MlemMiddleware PR](https://github.com/mlemgroup/MlemMiddleware/pull/66).

### CODE REVIEW

File: Mlem.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2787,8 +2795,8 @@
 			isa = XCRemoteSwiftPackageReference;
 			repositoryURL = "https://github.com/mlemgroup/MlemMiddleware";
 			requirement = {
-				kind = upToNextMinorVersion;
-				minimumVersion = 0.42.0;
+				branch = sjmarf/remove;
+				kind = branch;
```
Comment (EricBAndrews) [inline Mlem.xcodeproj/project.pbxproj]: Reminder

File: Mlem/App/Views/Shared/CommentView.swift [hunk 2]
```diff
@@ -72,7 +73,8 @@ struct CommentView: View {
                                 .frame(height: 10)
                                 .imageScale(.small)
                         } else {
-                            EllipsisMenu(size: 24) { comment.menuActions(commentTreeTracker: commentTreeTracker) }
+                            ellipsisMenus
+//                            EllipsisMenu(size: 24) { comment.menuActions(commentTreeTracker: commentTreeTracker) }
```
Comment (EricBAndrews) [inline Mlem/App/Views/Shared/CommentView.swift]: *nit*

Dead code?

**Review by EricBAndrews — CHANGES_REQUESTED** [review]
Code looks good!

It's a little strange to have "Spam"/"Troll"/"Abuse" options for restoring posts--we should either disable them or offer alternatives ("Accident" would be one, others aren't jumping to mind right now)

**Comment by Sjmarf** [comment]
Makes sense; I've removed the reason shortcuts on the restore sheet 👍

**Review by EricBAndrews — APPROVED** [review]
LGTM! 🚀

**Pull request merged**